### ISSUE TITLE
Guidance should be given regarding acceptable algorithms for including page navigation in reflowable EPUB publications that do not have statically paginated equivalents.

### ISSUE THREAD

**Issue by jasonjgw** [w3c/epub-specs#1801]
In the EPUB Accessibility 1.1 draft, section 3.4.1.2, the following statement appears:
"EPUB Creators MAY include page navigation in reflowable EPUB Publications without statically paginated equivalents."

However, no guidance is given (nor is there a reference to non-normative guidance elsewhere) as to acceptable algorithms for adding page navigation to such publications. In particular, where should the synthesized page break locators occur in a publication without any statically paginated equivalent? Should there be a maximum word limit per notional "page"? Should the algorithm respect block element (e.g., paragraph) boundaries? How "large" should each "page" be? It's hard to envisage this suggestion's being implemented in a manner that is valuable to users without some degree of guidance on how to carry it out. I expect authors/publishers would wish to automate it, hence the emphasis on acceptable/appropriate algorithms in this comment.

**Comment by mattgarrish** [comment]
Another task force in the EPUB WG is actually looking at this issue as it's a general usability question for publishers who want to provide static locations when there isn't a print equivalent.

If they develop recommendations, it's certainly something we could point to from the techniques, though.

**Comment by dauwhe** [comment]
See also #1542

**Comment by jasonjgw** [comment]
If an algorithm is proposed, I think it should be cited in EPUB Accessibility as an informative reference.

It isn't entirely clear whether this issue should be marked as a duplicate of #1542.

**Comment by mattgarrish** [comment]
Looking at this issue again, the quoted text is a bit confusing. It says you "MAY include page navigation in reflowable EPUB Publications without statically paginated equivalents," but it doesn't say if you have to meet the objectives if you do.

If you don't have to meet the objectives in these cases, then technically how you paginate is out of scope of the specification, although we could always include a note about the algorithm if it's done in time.

If you do have to meet the objective, we should make this clearer.

My recollection is that meeting the objectives in these cases is not required because it's more of a general usability issue if there is only a digital edition.

**Comment by jasonjgw** [comment]
It isn't clear to me what the usability advantage of having page break locators in a purely digital document is supposed to be. There are usually existing structures, such as block-level elements, by which one can navigate. These occur below the level of document sections (one could argue that sectional navigation alone isn't fine-grained enough).

Returning to a remembered place is another use case, but a bookmark mechanism seems better suited than page break locators to that purpose. While I don't have a settled opinion on the matter, I would tend to conclude that pagination is an artifact of print that we don't need to carry over to purely electronic documents. Thus it appears reasonable to give authors the option of including pagination in this case, but not to require it - perhaps not even to encourage it.

Much of the scholarly journal literature that I read may not have a printed and paginated form at all. Even if it does, the printed version may only have a small distribution (e.g., as printed conference proceedings given to attendees). It may not be justifiable to require page break locators in the electronic version that correspond to a printed text which almost nobody actually has or is likely to refer to.

**Comment by mattgarrish** [comment]
> It isn't clear to me what the usability advantage of having page break locators in a purely digital document is supposed to be.

General coordination amongst a group is all it would help, I believe. Absent static locators, no one group has an advantage over another in terms of finding a location below a heading which is why this isn't strictly an accessibility issue. Include them and there's a device-independent way to get closer to the destination for everyone.

Counting block elements probably still comes into play, of course, but it's easier right now to get to the second paragraph on page 120 if there are static locations, for example, than to tell someone to count 156 paragraphs into chapter 4.

It's not to say there won't one day be a better means of referencing into digital works, but hopes for a brighter future shouldn't influence what is needed today.

> It may not be justifiable to require page break locators in the electronic version that correspond to a printed text which almost nobody actually has or is likely to refer to.

There are a set of recommendations for when you should include page breaks based on the presence, use, and ease of creation, but they're never required.

**Comment by Doktorchen** [comment]
I authors want to improve such an access, they simply can add attributes id and tile with the same value to each paragraph and section.
If they want to help even more, they can provide an additional expanded list of contents with links to all these identifiable structures.

Because this uses only techniques from XML/XHTML/SVG, this is good implemented and can be combined with the 'EPUB Canonical Fragment Identifiers' (but for most people it would be simpler to reference content only by a combination of file names and identifiers of course).


Similar constructs with numbered paragraphs are available as well for some older (printed) books, which were/are often used for citation. This is more useful than a 'pagination' in case there is more than one edition for the work over the years or at the same time.

**Comment by avneeshsingh** [comment]
+1 to leave it to locators task force of EPUB 3 WG, we can provide a link to that document when it is complete.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-11-11-epub-a11y#section4) on 2021-11-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 4. Guidance should be given regarding acceptable algorithms for including page navigation in reflowable EPUB publications that do not have statically paginated: #1801.

_See github issue [epub-specs#1801](https://github.com/w3c/epub-specs/issues/1801)._




**Avneesh Singh:** this issue was already closed. But if any members feel we need to discuss further, we can discuss this now..  
… basically there was a request that we provide guidance on page numbers for digital only publications, but in the meantime this work is already being taken on by virtual locators TF.  

</details>

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-11-11-epub-a11y#section4) on 2021-11-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 4. Guidance should be given regarding acceptable algorithms for including page navigation in reflowable EPUB publications that do not have statically paginated: #1801.

_See github issue [epub-specs#1801](https://github.com/w3c/epub-specs/issues/1801)._




**Avneesh Singh:** this issue was already closed. But if any members feel we need to discuss further, we can discuss this now..  
… basically there was a request that we provide guidance on page numbers for digital only publications, but in the meantime this work is already being taken on by virtual locators TF.  

</details>

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-11-11-epub-a11y#section4) on 2021-11-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 4. Guidance should be given regarding acceptable algorithms for including page navigation in reflowable EPUB publications that do not have statically paginated: #1801.

_See github issue [epub-specs#1801](https://github.com/w3c/epub-specs/issues/1801)._




**Avneesh Singh:** this issue was already closed. But if any members feel we need to discuss further, we can discuss this now..  
… basically there was a request that we provide guidance on page numbers for digital only publications, but in the meantime this work is already being taken on by virtual locators TF.  

</details>

**Comment by jasonjgw** [comment]
This issue should remain open until the non-normative reference is added to the document, or it is concluded that this will not happen and an alternative decision is made.
